spectacles/runner.py

```diff
@@ -349,12 +349,12 @@ def validate_sql(
                 )
 
                 for dimension in project.iter_dimensions(errored=True):
-                    dimension.errors = [
-                        error
-                        for error in dimension.errors
-                        if not isinstance(error, SqlError)
-                        or (dimension.name, error.metadata["sql"]) not in target_sql
-                    ]
+                    for error in dimension.errors:
+                        if (
+                            isinstance(error, SqlError)
+                            and (dimension.name, error.metadata["sql"]) in target_sql
+                        ):
+                            error.ignore = True
```

Comment:
@joshtemple Can you talk me through what's happening here? I realised I'm not sure I follow the logic (even pre-change). Thanks!

Reply:
The idea is to ignore errors from dimensions with SQL that matches the SQL of dimensions on the target ref (incremental skips). Previously, we just deleted those errors entirely. Now, we keep them but mark them as ignored.